programs/fileio.c

```diff
@@ -514,6 +520,79 @@ static unsigned long long FIO_compressLzmaFrame(cRess_t* ress, const char* srcFi
 }
 #endif
 
+#ifdef ZSTD_LZ4COMPRESS
+static int FIO_LZ4_GetBlockSize_FromBlockId (int id) { return (1 << (8 + (2 * id))); }
+static unsigned long long FIO_compressLz4Frame(cRess_t* ress, const char* srcFileName, U64 const srcFileSize, int compressionLevel, U64* readsize)
+{
+    unsigned long long inFileSize = 0, outFileSize = 0;
+
+    LZ4F_preferences_t prefs;
+    LZ4F_compressionContext_t ctx;
+
+    LZ4F_errorCode_t const errorCode = LZ4F_createCompressionContext(&ctx, LZ4F_VERSION);
+    if (LZ4F_isError(errorCode)) EXM_THROW(31, "zstd: failed to create lz4 compression context");
+
+    memset(&prefs, 0, sizeof(prefs));
+
+    prefs.autoFlush = 1;
+    prefs.compressionLevel = compressionLevel;
+    prefs.frameInfo.blockMode = blockIndependent; /* stick to defaults for lz4 cli */
```

Comment:
Great work-around !

Indeed, using the old enums on newer versions of lz4frame is supposed to trigger deprecation warnings.
This is by design, though it could be disabled by using `-Wno-deprecated` with `gcc`
(note : there is no custom macro to do that).

However, there is a long term intention to remove the legacy enums from lz4frame at some point in the future, since they clobber namespace. Hence, we'd better comply with this long-term plan, and use new enums instead.

The version-conditioned `#define` replacement is a great solution which makes it possible to remain compatible with older versions.